src/main/java/org/spoken_tutorial/health/elasticsearch/threadpool/TaskProcessingService.java

```diff
@@ -104,47 +114,73 @@ public void queueProcessor() {
+                    .findByStatusOrderByRequestTimeAscWithNqueries(Config.STATUS_PENDING);
             if (qmnts == null) {
                 try {
                     Thread.sleep(Config.NO_TASK_SLEEP_TIME);
                     continue;
 
                 } catch (InterruptedException e) {
-
+                    logger.info("Interrupted");
+                    break;
                 }
             }
-
+            if (qmnts.size() != 0) {
+                logger.info("QueryResultSize:{}", qmnts.size());
+            }
             for (QueueManagement qmnt : qmnts) {
+                MDC.put("queueId", '#' + Long.toString(qmnt.getQueueId()));
                 logger.info("Queueing:{}", qmnt);
-                if (skippedDocuments.containsKey(qmnt.getDocumentId())) {
-
-                    continue;
-                }
+                try {
+                    if (skippedDocuments.containsKey(qmnt.getDocumentId())) {
+                        logger.info("skipDocument contains the DocumentID: {}", qmnt.getDocumentId());
 
-                String path = qmnt.getDocumentPath();
-                if (path.startsWith("https://")) {
-                    if (!isURLWorking(path)) {
+                        MDC.remove("queueId");
                         continue;
                     }
 
-                }
-                skippedDocuments.put(qmnt.getDocumentId(), System.currentTimeMillis());
-                getRunningDocuments().put(qmnt.getDocumentId(), System.currentTimeMillis());
-                qmnt.setStatus(Config.STATUS_QUEUED);
-                qmnt.setQueueTime(System.currentTimeMillis());
-                repo.save(qmnt);
+                    String path = qmnt.getDocumentPath();
+                    if (path.startsWith("https://")) {
+                        if (!isURLWorking(path)) {
+                            logger.info("The documentPath url is not working: " + path);
+                            qmnt.setStatus(Config.STATUS_PENDING);
+                            repo.save(qmnt);
+
+                            MDC.remove("queueId");
+                            continue;
+                        }
+
+                    }
+                    skippedDocuments.put(qmnt.getDocumentId(), System.currentTimeMillis());
+                    getRunningDocuments().put(qmnt.getDocumentId(), System.currentTimeMillis());
+                    qmnt.setStatus(Config.STATUS_QUEUED);
+                    qmnt.setQueueTime(System.currentTimeMillis());
+
+                    applicationContext.getAutowireCapableBeanFactory().autowireBean(qmnt);
+                    repo.save(qmnt);
+                    logger.info("{}", qmnt.getStatusLog());
+                    taskExecutor.submit(qmnt);
 
-                applicationContext.getAutowireCapableBeanFactory().autowireBean(qmnt);
-                taskExecutor.submit(qmnt);
-                count = count + 1;
+                    count = count + 1;
 
+                } catch (Exception e) {
```

Comment:
} catch (InterruptedException e) {
      logger.info("Interrupted");
      break;
} } catch (Exception e) {
        logger.error("Exception Error", e);
        continue;
}

Add catch (InterruptedException e) also before this catch. It will have logger and break statements